constraint/pkg/apis/templates/crd_schema.go

```diff
@@ -0,0 +1,60 @@
+package templates
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+)
+
+var ConstraintTemplateSchemas map[string]*schema.Structural
+
+func init() {
+	ConstraintTemplateSchemas = make(map[string]*schema.Structural)
+
+	// Ingest the constraint template CRD for use in defaulting functions
+	crdJSON, err := yaml.YAMLToJSONStrict([]byte(constraintTemplateCRDYaml))
+	if err != nil {
+		panic(errors.Wrap(err, "Failed to convert Constraint Template yaml to JSON"))
+	}
+
+	unCRD := unstructured.Unstructured{}
+	unCRD.UnmarshalJSON(crdJSON)
```

Comment:
We can't unmarshal directly to `constraintTemplateCRD` and save the middle step of converting to `unstructured`?

Reply:
It doesn't look like `CustomResourceDefinition` has an Unmarshal function.

[`apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal.go`](https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apis/apiextensions/v1/marshal.go) _does_ have unmarshalling logic, but it only supports doing so into a `JSONSchemaPropsOrArray`.

One idea:

I did the unmarshalling + `FromUnstructured()` because it made it easier to dive into the `versions:` and populate my `ConstraintTemplateSchemas` map.  I could probably simplify this logic by diving straight to `versions` in the unstructured and unmarshalling those JSON blobs straight into `apiextensionsv1.JSONSchemaProps`.  I'd then convert to `apiextensions.JSONSchemaProps` (apiextensions has no unmarshalling methods, only the versioned sub-packages) and create the `Structural` type for the map.

Reply:
I tried this^ and it's more trouble than it's worth.  B/c I use multiple properties of a `version` in my loop, it pays to convert to a `CustomResourceDefinition{}`.